sdk/translation/ai-translation-text/src/customClient.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { getClient, ClientOptions } from "@azure-rest/core-client";
+import * as coreRestPipeline from "@azure/core-rest-pipeline";
+import {
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest
+} from "@azure/core-rest-pipeline";
+import { TextTranslationClient } from "./generated/clientDefinitions";
+import {
+  TranslatorCredential,
+  TranslatorAuthenticationPolicy,
+  TranslatorAzureKeyAuthenticationPolicy
+} from "./authentication";
+import {
+  AzureKeyCredential,
+  TokenCredential
+} from "@azure/core-auth";
+
+const DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
+const DEFAULT_ENPOINT = "https://api.cognitive.microsofttranslator.com";
+const PLATFORM_HOST = "cognitiveservices";
+const PLATFORM_PATH = "/translator/text/v3.0";
+
+/**
+ * Initialize a new instance of `TextTranslationClient`
+ * @param endpoint type: string, Supported Text Translation endpoints (protocol and hostname, for example:
+ *     https://api.cognitive.microsofttranslator.com).
+ * @param options type: ClientOptions, the parameter for all optional parameters
+ */
+export default function createClient(
+  endpoint: undefined | string,
+  credential: undefined | TranslatorCredential | AzureKeyCredential | TokenCredential = undefined,
+  options: ClientOptions = {}
+): TextTranslationClient {
+  let serviceEndpoint: string;
+  if (!endpoint) {
+    serviceEndpoint = DEFAULT_ENPOINT;
+  } else if (endpoint.toLowerCase().indexOf(PLATFORM_HOST) != -1) {
+    serviceEndpoint = `${endpoint}${PLATFORM_PATH}`;
+  } else {
+    serviceEndpoint = endpoint;
+  }
+
+  const baseUrl = options.baseUrl ?? `${serviceEndpoint}`;
+
+  const userAgentInfo = `azsdk-js-ai-translation-text-rest/1.0.0-beta.1`;
+  const userAgentPrefix =
+    options.userAgentOptions && options.userAgentOptions.userAgentPrefix
+      ? `${options.userAgentOptions.userAgentPrefix} ${userAgentInfo}`
+      : `${userAgentInfo}`;
+  options = {
+    ...options,
+    userAgentOptions: {
+      userAgentPrefix,
+    },
+  };
+
+  const client = getClient(baseUrl, options) as TextTranslationClient;
+  client.pipeline.addPolicy(apiVersionPolicy);
+
+  if (credential instanceof TranslatorCredential) {
+    const mtAuthneticationPolicy = new TranslatorAuthenticationPolicy(credential as TranslatorCredential);
+    client.pipeline.addPolicy(mtAuthneticationPolicy);
+  } else if (credential instanceof AzureKeyCredential) {
+    const mtKeyAuthenticationPolicy = new TranslatorAzureKeyAuthenticationPolicy(credential as AzureKeyCredential);
+    client.pipeline.addPolicy(mtKeyAuthenticationPolicy);
+  } else if (credential) {
+    client.pipeline.addPolicy(
+      coreRestPipeline.bearerTokenAuthenticationPolicy({
```

Comment:
Looks like you support AAD already.

Reply:
AAD will be added for GA, we are not going to add any mention in preview as it was not validated.